scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,118 @@
+    # This element points to the 'v' in 'pv' -> and we map it to the
+    # corresponding property 'p'
+    val_prop_col = part_list[-1]
+
+    for part in part_list:
+      # set the base for special cases like median, etc.
+      if not measurement_assigned and 'measurement' in self.features.keys():
+        if part in self.features['measurement'].keys():
+          stat_var.update(self.features['measurement'][part])
+          measurement_assigned = True
+
+    # set the default statVar definition
+    if not measurement_assigned and 'measurement' in self.features.keys() and '_DEFAULT' in self.features['measurement'].keys():
+      stat_var.update(self.features['measurement']['_DEFAULT'])
+
+    # set the populationType attribute
+    stat_var['populationType'] = self._get_population_type(part_list)
+
+    # associate pvs to stat_var
+    if val_prop_col not in self.features['measurement'].keys():
+      if 'pvs' in self.features.keys():
+        #p = property, k = dictionary of substrings in column that points to a propertyValue
+        for p, k in self.features['pvs'].items():
+          for c, v in k.items():
+            ## check if the val_prop_col and the current key of the dict match 
+            if val_prop_col == c:
+              stat_var[p] = v
+              # add inferred properties from property 'p' to the stat_var
+              if 'inferredSpec' in self.features.keys() and p in self.features['inferredSpec'].keys():
+                stat_var.update(self.features['inferredSpec'][p])
+
+            ## if the above fails, check if the current key of dict matched with a token's substring
+            ## Eg: In table S2702, 'Uninsured' is the substring in the column that maps to the value 'NoHealthInsurance'.
+            ## This case is missed in the above condition and hence we check if the substring in each column token
+            ## In this case 'Total Uninsured' token is used to associate the pv of healthInsurance to the stat_var
+            for part in part_list:
+              if c in part:
+                stat_var[p] = v
+
+    #TODO: If quantityRanges occurs as values in multiple properties, find the property that relates to the column
+    #TODO: Add units and scalingFactor
+
+    # if the column is Margin of Error, update statType key
+    if 'Margin of Error' in part_list:
+      stat_var['statType'] = 'marginOfError'
+  
+    ## add Universe PVs based on the populationType of StatVar
+    dependent_properties = []
+    if 'universePVs' in self.features.keys():
+      for elem in self.features['universePVs']:
+        if stat_var['populationType'] == elem['populationType']:
+          # check for constraints of this populationType is in stat_var
+          for constraint in elem['constraintProperties']:
+            if constraint in stat_var.keys():
+              # if constraint and populationType satisties add dependent PVs
+              for k, v in elem['dependentPVs'].items():
+                stat_var[k] = v
```

Comment:
Shouldn't this already have happened in the "pvs" loop above?

Reply:
This did not happen in this case, I did not add the depdendentPVs under spec["pvs"] in the JSON. Thus they are not added in "pvs" loop. Is this approach fine?

Reply:
Hmmm.. if a certain PV is a dependent PV only for certain columns but not for others then it will get handled in the pvs loop above, which you might not always want?

So feels safer to add the DPV columns too to pvs?

Reply:
That is true, and I agree adding the DPV columns to pvs will be safer.
I have tested this and it seems to work without any issues, will add it in the next commit.